### PR TITLE
docs: modified some en statement in ref/arch/online_arch

### PR DESCRIPTION
**Pull request by michelle-qinqin**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
docs

* **What is the current behavior?** (You can also link to an open issue here)
+ the discription of disk-based storage engine is out of date
+ there are some translation errors like :“从节点” is translated to "from the node"

* **What is the new behavior (if this is a feature change)?**
fixed above questions

### CODE REVIEW

**Review by lumianph — CHANGES_REQUESTED** [review]
let us change all master/slave to leader/follower :-)

File: docs/en/reference/arch/online_arch.md [hunk 1]
```diff
@@ -2,46 +2,46 @@
...
+Nameserver is mainly used for tablet management and failover. When a tablet node goes down, the nameserver will be triggered to execute a series of tasks to perform a failover. When the tablet node recovers, nameserver will reload the data into this node. In order to ensure the high availability of nameserver itself, multiple nameserver instances will be deployed at the same time with the primary/secondary node deployment mode. Only one primary node can exist at the same time. The primary node will be determined from multiple nameserver instances preemptly through ZooKeeper. If the current primary node unexpectedly goes offline, remaining secondary nodes will use ZooKeeper to elect a new priamry node.
```
Comment (lumianph) [inline docs/en/reference/arch/online_arch.md]: The primary node will be determined from multiple nameserver instances preemptly through ZooKeeper
=>
The primary node will be determined from multiple nameserver instances by preemption through ZooKeeper

File: docs/en/reference/arch/online_arch.md [hunk 2]
```diff
@@ -2,46 +2,46 @@
...
 
 ## 4. Tablets
-Tablet is used by OpenMLDB to execute SQL and data storage, and it is also the core of the entire OpenMLDB function implementation and the bottleneck of resource consumption. From a functional point of view, Tablet includes two modules: SQL engine and storage engine. Tablet is also the smallest configurable granularity of OpenMLDB deployment resources. A tablet cannot be split across multiple physical nodes; however, there can be multiple tablets on a single physical node.
+Tablet is used by OpenMLDB to execute SQL and data storage. It is the core part of the entire OpenMLDB function implementation and the bottleneck of resource consumption. From the functional point of view, a tablet includes two modules: a SQL engine and a storage engine. Tablet is also the smallest configurable unit of OpenMLDB deployment resources. A tablet cannot be split across multiple physical nodes; however, there can be multiple tablets on a single physical node.
 
 ### 4.1 SQL Engine
 The SQL engine is responsible for executing SQL query calculations. The execution process after the SQL engine receives the request of the SQL query is shown in the following figure:
```
Comment (lumianph) [inline docs/en/reference/arch/online_arch.md]: The SQL engine is responsible for executing SQL queries.

File: docs/en/reference/arch/online_arch.md [hunk 3]
```diff
@@ -2,46 +2,46 @@
...
+The SQL engine parses SQL into an AST syntax tree through [ZetaSQL](https://github.com/4paradigm/zetasql). We have optimized the open source ZetaSQL to added special SQL syntax ( such as `LAST JOIN`, `WINDOW UNION`, etc) for feature engineering. After compiling, optimization, and LLVM-based codegen as shown in the figure above, the execution plan is finally generated. Then the SQL engine will access the storage layer data through the catalog and perform the final SQL calculation based on the execution plan. In the cluster version, a distributed execution plan will be generated, and the calculation task will be sent to other tablet nodes for to be exectued. Currently, the SQL engine  adopts the **push** mode, which distributes tasks to the nodes where the data is located for execution instead of pulling the data back. This has the benefit of reducing data transfers.
```
Comment (lumianph) [inline docs/en/reference/arch/online_arch.md]: perform the final SQL calculation => perform the final computation

and the calculation task will be sent to other tablet nodes for to be exectued =>
and the computing task will be sent to other tablet nodes to execute.

File: docs/en/reference/arch/online_arch.md [hunk 4]
```diff
@@ -2,46 +2,46 @@
...
-The Storage engine is responsible for the storage of OpenMLDB data and supports the corresponding high-availability-related functions.
+The Storage engine is responsible for the storage of data and supports the corresponding high-availability-related functions.
+
+:::{note}
+OpenMLDB supports two kinds storage engines: memory-based and disk-based. This article introduces the memory storage engine which is self-developed especially for OpenMLDB. The disk storage engine is based on RocksDB and has the same mechanism. See [RocksDB](https://github.com/facebook/rocksdb) for details.
+:::
 
 #### Data Distribution
-OpenMLDB Cluster Edition is a distributed database. The data of a table will be partitioned, and multiple copies will be created, and finally distributed in different nodes. Two important concepts are explained here: replicas and partitioning.
+OpenMLDB Cluster Edition is a distributed database. One table will be partitioned, and multiple copies will be created and finally distributed in different nodes. There are two important concepts: replicas and partitioning.
```
Comment (lumianph) [inline docs/en/reference/arch/online_arch.md]: Cluster Edition => cluster version

File: docs/en/reference/arch/online_arch.md [hunk 5]
```diff
@@ -2,46 +2,46 @@
...
+- Partition: When a table (or specifically a copy) is stored, it will be further divided into multiple partitions for distributed computing. The number of partitions can be specified when the table is created, but once created, the number of partitions cannot be dynamically modified. A partition is the smallest unit of master-slave synchronization, expansion and contraction of a storage engine. A partition can be flexibly migrated between different tablets. At the same time, different partitions of a table can be calculated in parallel, improving the performance of distributed computing. OpenMLDB will automatically try to balance the number of partitions on each tablet to improve the overall performance of the system. Multiple partitions of a table may be distributed on different tablets. Partitions are divided into primary partitions (leader) and secondary partition (followers). When a calculation request is received, the request will be sent to the primary partition corresponding to the data for calculation; the secondary partition is used to ensure high availability.
```
Comment (lumianph) [inline docs/en/reference/arch/online_arch.md]: calculation => computation

**Review by lumianph — CHANGES_REQUESTED** [review]

File: docs/en/reference/arch/online_arch.md (hunk 5 above)
Comment (lumianph) [inline docs/en/reference/arch/online_arch.md]: change all master/slave to leader/follower

Comment (michelle-qinqin) [reply docs/en/reference/arch/online_arch.md]: ok~

**Comment by michelle-qinqin** [comment]
fixed spelling mistakes: stoage engine => storage, namesever=>nameserver,
improve statements: master=>leader, slave=>follower, calculation => computation , etc

**Review by lumianph — APPROVED** [review]

**Pull request merged**